### PR TITLE
Add some clarifying language to the semantics of metric instrument naming

### LINKED ISSUE #434: Clarify name/version identity for named tracer/meters.
From [OTEP 76](https://github.com/open-telemetry/oteps/pull/76), Bogdan's comments:

>  is the name the key that identifies the Tracer/Meter or name + version? What is the expected behavior when (I don't know how is it possible) there are 2 requests to the getTracer with same name and different versions?

and

> I think it is important mainly for the metrics side. One important thing we need to ensure the name of the metrics are unique inside a "component" (instrumentation library), so if the component is identified by the name then I need to keep a list of all the registered instruments at the name level vs at the name+version level.

Please clarify this when including the related changes into the specification.

### PR DESCRIPTION
**Pull request by jkwatson**
Should hopefully let us close #434 .

### CODE REVIEW

File: specification/metrics/api-user.md [hunk 1]
```diff
@@ -55,27 +55,33 @@ Implementations SHOULD provide a single global default `MeterProvider`. The `get
...
+
 Metric instruments are defined using a `Meter` instance, using a variety
-of `New` methods specific to the kind of metric and type of input(integer
+of `New` methods specific to the kind of metric and type of input (integer
 or floating point).  The Meter will return an error when a metric name is
-already registered with a different kind for the same name.  Metric systems
-are expected to automatically prefix exported metrics by the namespace in a
-manner consistent with the target system.  For example, a Prometheus exporter
-SHOULD use the namespace followed by `_` as the
+already registered with a different kind for the same name. Metric systems
```
Comment (MrAlias) [inline specification/metrics/api-user.md:81]:
```suggestion
already registered with a different kind for the same name. Metric exporting systems
```

File: specification/metrics/api-user.md [hunk 2]
```diff
@@ -55,27 +55,33 @@ Implementations SHOULD provide a single global default `MeterProvider`. The `get
...
+are expected to automatically prefix exported metrics by the namespace, if
```
Comment (MrAlias) [inline specification/metrics/api-user.md:82]: We probably want to rephrase this to not prescribe a solution, but instead assign responsibility.

```suggestion
are responsible for disambiguation in the context of the namespace, if
```

File: specification/metrics/api-user.md [hunk 3]
```diff
@@ -55,27 +55,33 @@ Implementations SHOULD provide a single global default `MeterProvider`. The `get
...
-Metric names belong to a namespace, which is the `name` of the associated `Meter`,
+Metric instrument names belong to a namespace, which is the `name` of the associated `Meter`,
 allowing the same metric name to be used in multiple libraries of code,
 unambiguously, within the same application.
 
+Metric instrument names SHOULD be semantically meaningful, even when viewed
+outside of the context of the originating Meter name. For example, when instrumenting
+an http server library, "latency" is not an appropriate instrument name, as it is too generic.
+Instead, favor a name like "http_request_latency", as it informs the viewer of the
+semantic meaning of the latency.
```
Comment (bogdandrutu) [inline specification/metrics/api-user.md:76]: Just add an extra comment to say that this is an example and the list of standard names will be defined separately. I try to avoid making this example the canonical name for `http_request_latency` (not because it is a bad name, but because every document may include some examples and we want to have a semantic convention separate doc to keep all the names in one place).

File: specification/metrics/api-user.md [hunk 4]
```diff
@@ -55,27 +55,33 @@ Implementations SHOULD provide a single global default `MeterProvider`. The `get
...
-external systems.  Metric names conform to the following syntax:
+external systems.  Metric instrument names conform to the following syntax:
 
 1. They are non-empty strings
 2. They are case-insensitive
 3. The first character must be non-numeric, non-space, non-punctuation
 4. Subsequent characters must be belong to the alphanumeric characters, '_', '.', and '-'.
 
-Metric names belong to a namespace, which is the `name` of the associated `Meter`,
+Metric instrument names belong to a namespace, which is the `name` of the associated `Meter`,
```
Comment (bogdandrutu) [inline specification/metrics/api-user.md:68]: We discussed yesterday during metrics sig about this, I think the `meter name` is not a `namespace`, more like a `source`. Also I heard on gitter that you want to followup on this, what do you want to do with this PR then?

File: specification/metrics/api-user.md (hunk 3 above)
Comment (jkwatson) [reply specification/metrics/api-user.md:76]: I think until we have semantic conventions, we need some guidance, via examples, no?

File: specification/metrics/api-user.md (hunk 4 above)
Comment (jkwatson) [reply specification/metrics/api-user.md:68]: I'd like to leave this PR open; one way or another, clarification is needed.

**Comment by jmacd** [comment]
Posting @jkwatson's related notes on the Metrics SIG discussion: https://docs.google.com/document/d/1Y-Ck7NxRMEdnKueFFCVnYsdUIFeZ9EN4PpEFCkHu2-s/edit

File: specification/metrics/api-user.md (hunk 4 above)
Comment (arminru) [reply specification/metrics/api-user.md:68]: @bogdandrutu We adapted the spec to explicitly use the meter name as namespace a while ago after it was proposed in a SIG spec meeting since this puts the metric names into a namespace per library to avoid name clashes between libraries unaware of each other. See https://github.com/open-telemetry/opentelemetry-specification/issues/391#issuecomment-566299849, #408 and https://github.com/open-telemetry/oteps/pull/76#issuecomment-574846369. What has changed since then?

File: specification/metrics/api-user.md (hunk 3 above)
Comment (andrewhsu) [reply specification/metrics/api-user.md:76]: from the spec sig mtg today, sounds like the example just needs to be touched up so the example is not interpreted as authoritative cc @bogdandrutu @jkwatson

Comment (jkwatson) [reply specification/metrics/api-user.md]: I updated the language in here. PTAL

**Comment by jkwatson** [comment]
The failing CI check is passing for me. Anyone have any idea what's going on?

**Comment by jkwatson** [comment]
ok, CI is fixed.

File: specification/metrics/api-user.md (hunk 4 above)
Comment (bogdandrutu) [reply specification/metrics/api-user.md:66]: @arminru do you still have concerns?

Comment (arminru) [reply specification/metrics/api-user.md:66]: I thought you wanted to revise this and depart from using the meter name as a namespace / collision domain entirely but maybe I was missing context or misunderstood.
I'm fine with the current wording this PR introduces and therefore had already approved it.
